python/cudf/udf_cpp/nrt/_nrt_cuda.cuh

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <cuda/atomic>
+
+typedef __device__ void (*NRT_dtor_function)(void* ptr, size_t size, void* info);
+typedef __device__ void (*NRT_dealloc_func)(void* ptr, void* dealloc_info);
+
+typedef __device__ void* (*NRT_malloc_func)(size_t size);
+typedef __device__ void* (*NRT_realloc_func)(void* ptr, size_t new_size);
+typedef __device__ void (*NRT_free_func)(void* ptr);
+
+extern "C" {
+struct MemInfo {
+  cuda::atomic<size_t, cuda::thread_scope_device> refct;
+  NRT_dtor_function dtor;
+  void* dtor_info;
+  void* data;
+  size_t size;
+};
+}
+
+typedef struct MemInfo NRT_MemInfo;
+
+// Globally needed variables
+struct NRT_MemSys {
+  /* Shutdown flag */
+  int shutting;
```

Comment:
Can we get by without this? I don't think we need to worry about the interpreter shutting down when we're on-device.

Reply:
removed this